src/backend/distributed/utils/metadata_cache.c

```diff
@@ -677,6 +678,25 @@ InvalidateDistRelationCacheCallback(Datum argument, Oid relationId)
 			cacheEntry->isValid = false;
 		}
 	}
+
+	/*
+	 * If pg_dist_partition is being invalidated drop all state
+	 * This happens pretty rarely, but most importantly happens during
+	 * DROP EXTENSION citus;
+	 */
+
+	if (relationId != InvalidOid && relationId == distPartitionRelationId)
```

Comment:
Why not move this into the if-else block above?

    if (relationId == InvalidOid)
    ...
    else if (relationId == distPartitionRelationId)
    ...
    else